core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1815,6 +1838,22 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
                 // this method has been called by BlockChain for all relevant transactions. Otherwise we'd double
                 // count.
                 ignoreNextNewBlock.add(txHash);
+
+                // When a tx is received from the best chain, if other txns that spend this tx are IN_CONFLICT,
```

Comment:
I suggest to not extract a method here. We don't need this functionality anywhere else so far and Wallet has already an overwhelming amount of methods. Having said that, this is not life or dead to me, so if you insist, I will extract the method.